fili-core/src/main/java/com/yahoo/bard/webservice/logging/LogFormatterProvider.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.logging;
+
+import com.yahoo.bard.webservice.config.SystemConfig;
+import com.yahoo.bard.webservice.config.SystemConfigProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A class to provide a log formatter instance.
+ */
+public class LogFormatterProvider {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LogFormatterProvider.class);
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
+
+    private static final String DEFAULT_LOG_FORMATTER_IMPL = JsonLogFormatter.class.getCanonicalName();
+
+    private static final String LOG_FORMATTER_IMPL_KEY = SystemConfigProvider.getInstance().getStringProperty(
+            SYSTEM_CONFIG.getPackageVariableName("log_formatter_impl"),
+            DEFAULT_LOG_FORMATTER_IMPL
+    );
+
+    /**
+     * The instance of the Log Formatter.
+     */
+    private static final LogFormatter LOG_FORMATTER = getInstance();
+
+    /**
+     *  Get an instance of LogFormatter.
+     *
+     *  @return an instance of LogFormatter
+     */
+    public static LogFormatter getInstance() {
+        if (LOG_FORMATTER != null) {
+            return LOG_FORMATTER;
+        }
+        String logFormatterImplementation = System.getenv(LOG_FORMATTER_IMPL_KEY);
```

Comment:
This construction allows for the `LOG_FORMATTER` singleton holder to be unset if somehow something calls `getInstance` before this class does. It _shouldn't_ happen, but it would be good if we can insulate ourselves. A better construction is something like this:

```java
public String getInstance() {
    if (MY_INSTANCE == null) {
        // create myInstance
        MY_INSTANCE = myInstance;
    }
    return MY_INSTANCE;
}
```

By having only 1 return path, it's easier to reason about having set the singleton holder.

And if you want to get even more paranoid, you can have the body of that `if` be it's own `synchronized` helper method (which should also check if it's already set).